cpp/benchmarks/common/generate_input.cu

```diff
@@ -206,42 +274,35 @@ struct random_value_fn<T, std::enable_if_t<!std::is_same_v<T, bool> && cudf::is_
   {
   }
 
-  T operator()(std::mt19937& engine)
+  auto operator()(thrust::minstd_rand& engine, unsigned size)
   {
     // Clamp the generated random value to the specified range
-    return std::max(std::min(dist(engine), upper_bound), lower_bound);
+    // return std::max(std::min(dist(engine), upper_bound), lower_bound);
+    return dist(engine, size);
   }
 };
 
 /**
  * @brief Creates an boolean value with given probability of returning `true`.
  */
 template <typename T>
-struct random_value_fn<T, std::enable_if_t<std::is_same_v<T, bool>>> {
-  std::bernoulli_distribution b_dist;
-
-  random_value_fn(distribution_params<bool> const& desc) : b_dist{desc.probability_true} {}
-  bool operator()(std::mt19937& engine) { return b_dist(engine); }
-};
-
-size_t null_mask_size(cudf::size_type num_rows)
-{
-  constexpr size_t bitmask_bits = cudf::detail::size_in_bits<cudf::bitmask_type>();
-  return (num_rows + bitmask_bits - 1) / bitmask_bits;
-}
-template <typename T>
-void set_element_at(T value,
-                    bool valid,
-                    std::vector<T>& values,
-                    std::vector<cudf::bitmask_type>& null_mask,
-                    cudf::size_type idx)
-{
-  if (valid) {
-    values[idx] = value;
-  } else {
-    cudf::clear_bit_unsafe(null_mask.data(), idx);
+struct random_value_fn<T, typename std::enable_if_t<std::is_same_v<T, bool>>> {
+  // bernoulli_distribution
```

Comment:
This is slightly misleading since `dist` is normally distributed but we are using it to generate a Bernoulli distribution for the output random value. I would move this comment to the struct's comment if you think it's important.

Reply:
Oh never mind, I think I was confused by the placement of @vuule's comments below and didn't realize how `dist` was actually being initialized. `dist` is in fact Bernoulli, ignore my comment.